expr/func_keeplastvalue.go

```diff
@@ -0,0 +1,83 @@
+package expr
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/grafana/metrictank/api/models"
+	schema "gopkg.in/raintank/schema.v1"
+)
+
+type FuncKeepLastValue struct {
+	in    GraphiteFunc
+	limit int64
+}
+
+func NewKeepLastValue() GraphiteFunc {
+	return &FuncKeepLastValue{limit: math.MaxInt64}
+}
+
+func (s *FuncKeepLastValue) Signature() ([]Arg, []Arg) {
+	var stub string
+	return []Arg{
+			ArgSeriesList{val: &s.in},
+			ArgIn{key: "limit",
+				opt: true,
+				args: []Arg{
+					ArgInt{val: &s.limit},
+					// Allow user to specify 'INF' as value. if so, will fall back to maxInt
+					// Ignores any other strings (just like Graphite)
+					ArgString{val: &stub},
```

Comment:
Don't you mean 'inf' not 'INF' ?
this comment is confusing because it implies the behavior for INF vs other strings is different,
but actually in all cases (inf, INF, or any other strings) it will always just use the default of MaxInt64 because *ANY* string value is ignored. (not just "any other string" as it states)

Reply:
Graphite actually says that "INF" is infinity, but accepts any string since it doesn't check for types.
From graphite:
`Param('limit', ParamTypes.integer, default='INF')`

I think the way this function is implemented in graphite is not what they intended it to be, but I'm including the option to put in a string just to mimic the behavior.